exasol_script_languages_developer_sandbox/lib/export_vm/run_export_vm.py

```diff
@@ -0,0 +1,112 @@
+    export_image_task_id = \
+        aws_access.export_ami_image_to_vm(image_id=ami_id, tag_value=tag_value,
+                                          description="VM Description", role_name=vmimport_role,
+                                          disk_format=vm_image_format,
+                                          s3_bucket=vm_bucket, s3_prefix=bucket_prefix)
+
+    export_image_task = aws_access.get_export_image_task(export_image_task_id)
+
+    logging.info(
+        f"Started export of vm image to {vm_bucket}/{bucket_prefix}. "
+        f"Status message is {export_image_task.status_message}.")
+    last_progress = export_image_task.progress
+    last_status = export_image_task.status
+    while export_image_task.is_active:
+        time.sleep(config.global_config.time_to_wait_for_polling)
+        export_image_task = aws_access.get_export_image_task(export_image_task_id)
+        if export_image_task.progress != last_progress or export_image_task.status != last_status:
+            logging.info(f"still running export of vm image to {vm_bucket}/{bucket_prefix}. "
+                         f"Status message is {export_image_task.status_message}. "
+                         f"Progess is '{export_image_task.progress}'")
+        last_progress = export_image_task.progress
+        last_status = export_image_task.status
+    if not export_image_task.is_completed:
+        raise RuntimeError(f"Export of VM failed: status message was {export_image_task.status_message}")
+
+
+def create_ami(aws_access: AwsAccess, ami_name: str, tag_value: str, instance_id: str) -> str:
+    """
+    Creates a new AMI with the given name (parameter ami_name) for the EC2-Instance identified by parameter instance_id.
+    The AMI will be tagged with given tag_value.
+    :raises RuntimeError if an error occured during creation of the AMI
+    Returns the ami_id if the export-image-task succeeded.
+    """
+    logging.info(f"create ami with name '{ami_name}' and tag(s) '{tag_value}'")
+    ami_id = aws_access.create_image_from_ec2_instance(instance_id, name=ami_name, tag_value=tag_value,
+                                                       description="Image Description")
+
+    ami = aws_access.get_ami(ami_id)
+    while ami.is_pending:
+        logging.info(f"ami  with name '{ami.name}' and tag(s) '{tag_value}'  still pending...")
+        time.sleep(config.global_config.time_to_wait_for_polling)
+        ami = aws_access.get_ami(ami_id)
+    if not ami.is_available:
+        raise RuntimeError(f"Failed to create ami! ami state is '{ami.state}'")
+    return ami_id
+
+
+def export_vm(aws_access: AwsAccess,
+              instance_id: str,
+              vm_image_formats: Tuple[str, ...],
+              asset_id: AssetId) -> None:
+    vm_bucket = find_vm_bucket(aws_access)
+    vmimport_role = find_vm_import_role(aws_access)
+    tag_value = asset_id.tag_value
+    bucket_prefix = f"{asset_id.bucket_prefix}/"
+    has_errors = False
+    try:
+        try:
```

Comment:
This function reads very hard, maybe try to simplify by  un-nesting it.

Reply:
I opened #21 to change that.